### PR TITLE
Python bindings upd

### PR DESCRIPTION
**Pull request by ataymano**
Python bindings. Client lib fixes (actions ordering, chosen action setting, json output).

### CODE REVIEW

**Comment by JohnLangford** [comment]
This pull request **introduces 1 alert** when merging 53941c6f0d2a37aac43f3bf9b7afd8058492254e into d72a47871b6f4e0170a59050f0e7f99c3439778d - [view on LGTM.com](https://lgtm.com/projects/g/JohnLangford/vowpal_wabbit/rev/pr-aa335fc4ea8ba20cfd3dca7580c94d8c31ba22c9)

**new alerts:**

* 1 for Variable defined multiple times

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by marco-rossi29** [comment]
We have the seg fault on travis build that I discovered last week. I think someone should take a look

**Comment by JohnLangford** [comment]
This pull request **introduces 2 alerts** when merging 5349a1ab7fd2b7586fefd157b4ab4fc50bf45b01 into d72a47871b6f4e0170a59050f0e7f99c3439778d - [view on LGTM.com](https://lgtm.com/projects/g/JohnLangford/vowpal_wabbit/rev/pr-739c7d17c93744213be312c8c964f1b9607ede1d)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by JohnLangford** [comment]
This pull request **introduces 2 alerts** when merging b66d8aba33e0b62c76465c95535a73ad64d12ba2 into d72a47871b6f4e0170a59050f0e7f99c3439778d - [view on LGTM.com](https://lgtm.com/projects/g/JohnLangford/vowpal_wabbit/rev/pr-25f06261610abf18657ce2c6c38857b2798d6d0e)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: reinforcement_learning/doc/mainpage.dox [hunk 1]
```diff
@@ -23,7 +23,7 @@ reinforcement_learning::live_model class is the interface to RL Inference API
 \snippet examples/basic_usage_cpp/basic_usage_cpp.cc (2) Initialize the API
 - choose_rank() to choose an action from a list of actions
 \snippet examples/basic_usage_cpp/basic_usage_cpp.cc (3) Choose an action
-- (Optional) report_outcome() to provide feedback on choosen action  
+- (Optional) report_outcome() to provide feedback on chosen action
```
Comment (jackgerrits) [inline reinforcement_learning/doc/mainpage.dox]: 👍

**Pull request closed without merging**